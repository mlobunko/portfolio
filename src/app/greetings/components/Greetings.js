import React from "react";
import ReactRevealText from "react-reveal-text";

import {
  StyledArrow,
  StyledBottom,
  StyledBottomText,
  StyledGreetings,
  StyledHeader,
  StyledHeaderFirstSentence,
  StyledHeaderSecondSentence
} from "./styles";
import ImageGreetings from "../images/image-greetings.jpg";

export class Greetings extends React.Component {
  state = {
    show: false
  };

  componentDidMount() {
    setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
  }

  render() {
    return (
      <StyledGreetings image={ImageGreetings}>
        <StyledHeader>
          <StyledHeaderFirstSentence>
            <ReactRevealText
              show={this.state.show}
              transitionTime={2000}
              delayMin={0}
              delayMax={1000}
            >
              Hi! I'm Mika.
            </ReactRevealText>
          </StyledHeaderFirstSentence>
          <StyledHeaderSecondSentence>
            <ReactRevealText
              show={this.state.show}
              transitionTime={2000}
              delayMin={1000}
              delayMax={2000}
            >
              I build web applications.
            </ReactRevealText>
          </StyledHeaderSecondSentence>
        </StyledHeader>
        <StyledBottom>
          <StyledBottomText>
            <ReactRevealText
              show={this.state.show}
              transitionTime={2000}
              delayMin={2000}
              delayMax={3000}
            >
              Learn more about what I do
            </ReactRevealText>
          </StyledBottomText>
          <StyledArrow>
            <ReactRevealText
              show={this.state.show}
              transitionTime={2000}
              delayMin={2000}
              delayMax={3000}
            >
              Ë…
            </ReactRevealText>
          </StyledArrow>
        </StyledBottom>
      </StyledGreetings>
    );
  }
}
Greetings.displayName = "Greetings";

export default Greetings;
