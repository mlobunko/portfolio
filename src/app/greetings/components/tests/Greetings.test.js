import React from "react";
import { shallow } from "enzyme";

import { Greetings } from "../Greetings";

beforeEach(() => {
  jest.resetAllMocks();
  jest.useFakeTimers();
});

afterEach(() => {
  jest.clearAllTimers();
});

describe("render", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Greetings />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("methods", () => {
  it("setState call with right argumetns in setTimeOut on componentDidMount", () => {
    const wrapper = shallow(<Greetings />);
    wrapper.setState = jest.fn();
    wrapper.update();
    expect(wrapper.setState).toHaveBeenCalledTimes(0);
    jest.runAllTimers();
    expect(wrapper.setState).toHaveBeenCalledWith({ show: true });
  });
});
