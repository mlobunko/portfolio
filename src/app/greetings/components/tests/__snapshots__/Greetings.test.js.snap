// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledGreetings
  image="image-greetings.jpg"
>
  <StyledHeader>
    <StyledHeaderFirstSentence>
      <ReactRevealText
        className=""
        delayMax={1000}
        delayMin={0}
        show={false}
        style={Object {}}
        text=""
        threshold={0.2}
        timingFunction="linear"
        transitionTime={2000}
      >
        Hi! I'm Mika.
      </ReactRevealText>
    </StyledHeaderFirstSentence>
    <StyledHeaderSecondSentence>
      <ReactRevealText
        className=""
        delayMax={2000}
        delayMin={1000}
        show={false}
        style={Object {}}
        text=""
        threshold={0.2}
        timingFunction="linear"
        transitionTime={2000}
      >
        I build web applications.
      </ReactRevealText>
    </StyledHeaderSecondSentence>
  </StyledHeader>
  <StyledBottom>
    <StyledBottomText>
      <ReactRevealText
        className=""
        delayMax={3000}
        delayMin={2000}
        show={false}
        style={Object {}}
        text=""
        threshold={0.2}
        timingFunction="linear"
        transitionTime={2000}
      >
        Learn more about what I do
      </ReactRevealText>
    </StyledBottomText>
    <StyledArrow>
      <ReactRevealText
        className=""
        delayMax={3000}
        delayMin={2000}
        show={false}
        style={Object {}}
        text=""
        threshold={0.2}
        timingFunction="linear"
        transitionTime={2000}
      >
        Ë…
      </ReactRevealText>
    </StyledArrow>
  </StyledBottom>
</StyledGreetings>
`;
